bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/TapoConnector.java

```diff
@@ -0,0 +1,552 @@
+            }
+
+            setCipher(encryptedKey);
+            this.cookie = response.getResponseHeader("Set-Cookie").split(";")[0];
+            return true;
+        } catch (Exception ex) {
+            logger.warn("Something went wrong: {}", ex.getMessage());
+            tapoError.raiseError(ex, "could not create handshake");
+            this.cookie = "";
+            return false;
+        }
+    }
+
+    /**
+     * Create Cipher ( Decode Handshake )
+     *
+     * @param key encrypted key
+     * @return true if success
+     */
+    private Boolean setCipher(String key) {
+        logger.debug("Will try to decode the following key: {} ", key);
+
+        MimeEncode mimeEncode = new MimeEncode();
+
+        try {
+            byte[] decode = mimeEncode.decode(key.getBytes("UTF-8"));
+            byte[] decode2 = mimeEncode.decode(credentials.getPrivateKeyBytes());
+            Cipher instance = Cipher.getInstance("RSA/ECB/PKCS1Padding");
+            KeyFactory kf = KeyFactory.getInstance("RSA");
+            PrivateKey p = kf.generatePrivate(new PKCS8EncodedKeySpec(decode2));
+            instance.init(Cipher.DECRYPT_MODE, p);
+            byte[] doFinal = instance.doFinal(decode);
+            byte[] bArr = new byte[16];
+            byte[] bArr2 = new byte[16];
+            System.arraycopy(doFinal, 0, bArr, 0, 16);
+            System.arraycopy(doFinal, 16, bArr2, 0, 16);
+            this.tapoCipher = new TapoCipher(bArr, bArr2);
+            return true;
+        } catch (Exception ex) {
+            logger.warn("Something went wrong: {}", ex.getMessage());
+            tapoError.raiseError(ex);
+            return false;
+        }
+    }
+
+    /**
+     * Decrypt Response
+     * 
+     * @param responseBody encrypted string from response-body
+     * @return String decrypted responseBody
+     */
+    private String decryptResponse(String responseBody) {
+        try {
+            JsonObject jsonObject = gson.fromJson(responseBody, JsonObject.class);
+            String encryptedResponse = jsonObject.getAsJsonObject("result").get("response").getAsString();
+            String decryptedResponse = tapoCipher.decode(encryptedResponse);
+            return decryptedResponse;
+        } catch (Exception ex) {
+            return "";
+        }
+    }
+
+    /***********************************
+     *
+     * HTTP ACTIONS
+     *
+     ************************************/
+
+    /**
+     * login
+     *
+     * @return true if success
+     */
+    public boolean login() {
+        logger.debug("sending login");
+
+        String securePassthroughPayload = "";
+        tapoError.reset(); // reset ErrorHandler
+
+        /* ping device */
+        if (!pingDevice()) {
+            tapoError.raiseError(ERROR_DEVICE_OFFLINE, "device offline while login");
+            logout();
+            return false;
+        }
+
+        /* create handschake (cookie) */
+        this.createHandshake();
+
+        if (cookie != "") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.